plugins/balloonpanel/plugin.js

```diff
@@ -424,12 +497,29 @@
 
 				// This method will modify elementRect if the element is outside of allowedRect / editorRect.
 				// If it's outside then in
-				this._adjustElementRect( elementRect, isInline ? allowedRect : editorRect );
+				if ( selectionRects ) {
+					CKEDITOR.tools.array.forEach( selectionRects, function( item ) {
+						this._adjustElementRect( item, isInline ? allowedRect : editorRect );
+					}, this );
+				} else {
+					this._adjustElementRect( elementRect, isInline ? allowedRect : editorRect );
+				}
 
 				// The area of the panel.
 				var panelArea = panelWidth * panelHeight,
-					alignments = this._getAlignments( elementRect, panelWidth, panelHeight ),
-					minDifferenceAlignment, alignmentRect, areaDifference;
+					alignments, minDifferenceAlignment, alignmentRect, areaDifference;
+
+				if ( selectionRects ) {
```

Comment:
Can't this conditional be merged with the previous one? There are divided only by defining some variables.